lib/SILGen/SILGenConvert.cpp

```diff
@@ -654,6 +654,40 @@ ManagedValue SILGenFunction::emitExistentialErasure(
     return emitManagedRValueWithCleanup(existential);
   }
   case ExistentialRepresentation::Opaque: {
+  
+    // If the concrete value is a pseudogeneric archetype, first erase it to
+    // its upper bound.
+    auto anyObjectProto = getASTContext()
+      .getProtocol(KnownProtocolKind::AnyObject);
+    auto anyObjectTy = anyObjectProto
+      ? anyObjectProto->getDeclaredType()->getCanonicalType()
```

Comment:
Do we actually expect anyObjectProto to ever be null? Wouldn't the code crash below because of a null CanType?